src/mmw/js/src/user/views.js

```diff
@@ -51,42 +55,113 @@ var LoginModalView = Marionette.ItemView.extend({
         }
     },
 
-    setFields: function() {
+    // Primary Action might be "log in" or "sign up" or "reset password".
+    // By default this serializes all form data and POSTs to child's `url`.
+    // Override if other behavior is required.
+    primaryAction: function() {
+        var formData = this.$el.find('form').serialize();
+        this.model
+            .fetch({
+                method: 'POST',
+                data: formData
+            })
+            .done(_.bind(this.handleServerSuccess, this))
+            .fail(_.bind(this.handleServerFailure, this));
+    },
+
+    handleServerSuccess: function() {
         this.model.set({
-            username: $(this.ui.username.selector).val(),
-            password: $(this.ui.password.selector).val()
-        }, { silent: true });
+            'success': true,
+            'client_errors': null,
+            'server_errors': null
+        });
+    },
+
+    // Default failure handler, will display any `errors` received from server.
+    handleServerFailure: function(response) {
+        var server_errors = ["Server communication error"];
+        if (response.responseJSON.errors) {
+            server_errors = response.responseJSON.errors;
+        }
+        this.model.set({
+            'success': false,
+            'client_errors': null,
+            'server_errors': server_errors
+        });
     },
 
+    // Dismiss Action can be "cancel" or "continue as guest". Noop by default.
+    // Override if other behavior is required.
+    dismissAction: function() { },
+
+    // Performs Primary Action if model is in valid state.
+    // `setFields` must be defined on child.
     validate: function() {
         this.setFields();
+        if (this.model.isValid()) {
+            this.primaryAction();
+        }
+    }
+});
 
-        var errors = this.model.validate(this.model.attributes);
+var LoginModalView = ModalBaseView.extend({
+    template: loginModalTmpl,
 
-        if (!errors) {
-            this.login();
-        }
+    ui: {
+        username: '#username',
+        password: '#password',
+        signUp:   '.sign-up',
+        forgot:   '.forgot'
+    },
+
+    events: {
+        'click @ui.signUp': 'signUp',
+        'click @ui.forgot': 'forgot'
+    },
+
+    initialize: function(options) {
+        ModalBaseView.prototype.initialize(this, options);
```

Comment:
Interesting technique. Typically inheritance chains will call super methods with `call`/`apply`. Ex. `ModalBaseView.prototype.initialize.call(this, options);`